### ISSUE TITLE
Event listener timeout

### ISSUE THREAD

**Issue by caesarisme** [gram-js/gramjs#494]
I have noticed that when you subscribe to updates after some time (several hours I guess) it looses connection and new events are not being received. Cron task with restarting connection every 2 hours fixes the problem, but not completely. Any workaround there?

**Comment by Ali559** [comment]
Hello @caesarisme, 
I haven't noticed the same issue as your code, but I would suggest you call an API before registering the event listener like so 

```
await client.getMe();

// Then addEventHandler

```

**Comment by ShaunLWM** [comment]
Just realized I have this issue as well. Seemed like it's an old issue - probably Telegram servers themselves?

https://github.com/gram-js/gramjs/issues/280

My workaround is to set a cronjob to call `getMe()` every hour. Gonna see if it works.

**Comment by b10c77** [comment]
@caesarisme @ShaunLWM Have you been able to find a solution to this issue?

**Comment by ShaunLWM** [comment]
@b10c77 Nope. I just do a cron job to restart the service every 6 hours. Working well till now.

**Comment by sahilnarain** [comment]
> @b10c77 Nope. I just do a cron job to restart the service every 6 hours. Working well till now.

Though it's not clearly documented anywhere, a lot of developers have observed that a high-level method needs to be called to keep the listener alive.

From what I understand, this is on Telegram's API and not a matter of implementation in GramJS or any other client libraries.

I was facing the same problems, but instead of restarting my process again (which leads to missed messages if connect() call takes time) - I have implemented an asynchronous keep-alive mechanism. It looks something like this -


```
const keepAlive = async () => {
  setInterval( async () => {
    if(!client.connected) {
      await client.connect()
    }

    if(client.checkAuthorization()) {
      await client.getMe() 
      console.log('[Keep Alive]');
    }
  }, 30000);
};

keepAlive();
```

**Comment by pedroota** [comment]
Is it still working? Because I'm facing the exact same problem and the keepAlive function is not solving it

**Comment by Demigoddd** [comment]
> [@b10c77](https://github.com/b10c77) Nope. I just do a cron job to restart the service every 6 hours. Working well till now.

Hello, I would like to see the code of your solution to fix this problem.
